### ISSUE TITLE
poolmotortv generates extra hardware communication during scan

### ISSUE THREAD

**Issue by dschick** [sardana-org/sardana#1169]
Hej,

I have added some PoolMotor widgets in my taurus gui and it seems, that I do get extra hardware communication in a scan from that, but with motors that I do not scan.
So in case the taurus gui is closed I do not see the hardware communication logs and otherwise I do.

Is that possible?

Best 

Daniel

**Comment by teresanunez** [comment]
Hi @dschick,
                    I would say that you have the hardware communication produced by the read out of
the motor ... but this happens with or without running the scan. If you close the taurus gui you don't
have it any more because you don't have any client reading from your hardware.

                               Regards,
                                                  Teresa

**Comment by dschick** [comment]
thanks, I agree it is just the reading of the position.
It is actually only logged for hardware that is NOT Tango based.
I have the feeling that the reading of the motors happens just more frequent when I am doing a scan.
Is there something like a default reading period, like in jive?

**Comment by teresanunez** [comment]
Yes, there is a default reading period, and it is possible to change this period for your GUIs, but
 I better let the people of Alba go into the details.

**Comment by reszelaz** [comment]
Sorry for the late reply!

The PMTV widget listens to Tango events. These are sent only when the motion is handled by Sardana e.g. mv macro. Apart from that it also enables polling. This was done for the cases when motion was not issued from Sardana, e.g. moved from the intermediate control layer e.g. using a Tango device (not Sardana) or from the manufacturer control application, etc. This polling should only work when the widget is visible. Now I´m not sure about the polling period, would need to check it...

For the cases when motion is only done from the Sardana level, this polling may be unnecessary or even disturbing. So in my opinion we need some easy way to disable it.

**Comment by cpascual** [comment]
Taurus widgets activate the polling by default unless the Device declares that it will emit events. The polling is **not** disabled even if the widget is hidden (this is a design decision). 

But taurus optimizes the polling by doing a single `read_attributes()` call per device per poll period regradless  of how many widget are listening to that device attributes.

Because of this, customizing the PMTV widget to stop the polling makes little sense because as soon as some other widget access the same device, the poll would be done anyway.

IMHO, trying to optimize this not worth the effort considering how fragile the result would be.
